yarn.config.cjs

```diff
@@ -488,20 +490,23 @@ async function expectWorkspaceLicense(workspace) {
 function expectCorrectWorkspaceExports(workspace) {
   // All non-root packages must provide the location of the ESM-compatible
   // JavaScript entrypoint and its matching type declaration file.
-  expectWorkspaceField(workspace, 'exports["."].import', './dist/index.mjs');
   expectWorkspaceField(
     workspace,
-    'exports["."].types',
-    './dist/types/index.d.ts',
+    'exports["."].import.types',
```

Comment:
Note that `types` must be above `default`. I don't think Yarn enforces this currently, but we could look into doing so.